feat(docs): set up noir contracts in getting-started (#1770)
AztecProtocol/aztec-packages: docs/docs/dev_docs/getting_started/noir_contracts.md

---

> title: Setting up Aztec.nr Contracts

Reviewer: Would probably remove the setting up, we are already at a getting started.
```suggestion
title: Aztec.nr Contracts
```

---

## Introduction

This guide explains the necessary set up required to write your own contract using the Aztec.nr library and deploy it on the sandbox

:::info Prerequisite reading
If you haven't read [Aztec Sandbox](./sandbox.md) and [Noir](./noir.md), we recommend going there first.
:::

### Dependencies

> 1. You will need the Noir build tool `nargo`, which you can install via [`noirup`](https://github.com/noir-lang/noirup). Make sure you install the `aztec` version of nargo:

Reviewer 2: ```suggestion
1. You will need Noir's build tool `nargo`, which you can install via [`noirup`](https://github.com/noir-lang/noirup). Make sure you install the `aztec` version of nargo:
```

> 1. You will need the Noir build tool `nargo`, which you can install via [`noirup`](https://github.com/noir-lang/noirup). Make sure you install the `aztec` version of nargo:

Reviewer 2: nit: I would alter this section such that the bullet points are supported by sentences, rather than being sentences.

1. Nargo
_nargo install instructions ..._

2. Sandbox
_Sandbox installation instructions / link_

3. Starter repo
_Link to starter repo_

```bash
curl -L https://raw.githubusercontent.com/noir-lang/noirup/main/install | bash
noirup -v aztec
```

> 2. [Have a running sandbox and a code repo to interact with it](./sandbox.md)

Reviewer: Would probably use a small header rather than the list. When looking over the steps it "vanishes in the crowd". Also would benefit by https://github.com/AztecProtocol/aztec-packages/pull/1770#discussion_r1304810188.

## Set up for aztec.nr contracts
1. Inside the yarn project you created from the [Sandbox](./sandbox.md) page, create a sub-folder where the contracts will reside.
```bash
mkdir contracts
```

All contract projects will reside within this folder.

2. Next, create a noir project using nargo by running the following in the terminal from the `contracts` folder
```bash
cd contracts
nargo new example_contract
```

This creates a noir project with Nargo.toml (which is the manifest file of the project). Also, you will see a file in `example_contract/src/main.nr` which is where we will write our contract. 

Your folder should look like:
```
.
|-contracts
| |--example_contract
| |  |--src
| |  |  |--main.nr
|-src
| |--index.ts
```

> But before writing the contracts, we must add the aztec.nr library that adds smart contract syntax to Noir that the aztec sandbox can understand.

Reviewer 2: Making tone a bit more authoritative by removing but. 
```suggestion
Before writing the contracts, we must add the aztec.nr library. This adds smart contract utility functions for interacting with the Aztec network.
```

consistency as refered to Aztec network at bottom of the page, which I prefer to sandbox. These docs should be independent of the target network

3. Build the aztec.nr library on your machine.

> For this, clone the `aztec-packages` repository (which is the monorepo where the library resides) and build it anywhere on your machine

Reviewer: For sure. Removing step 3 entirely, and adding the git repo in toml should be the way to go.

> For this, clone the `aztec-packages` repository (which is the monorepo where the library resides) and build it anywhere on your machine

Reviewer: ```suggestion
For this, clone the `aztec-packages` repository (which is the monorepo where the library resides) and build it anywhere on your machine.
```

> For this, clone the `aztec-packages` repository (which is the monorepo where the library resides) and build it anywhere on your machine

Author: updated -> nargo.toml can now directly talk to a directory in a gh repo!

```bash
cd path/to/where/you/want/this/package
```

```bash
git clone git@github.com:AztecProtocol/aztec-packages.git
cd aztec-packages
./bootstrap.sh
```
`./boostrap.sh` is a script that builds the package.

4. Add aztec.nr library as a dependency to your noir project. Open Nargo.toml that is in the `contracts/example_contract` folder, and add the dependency section as follows:
```
[package]
name = "example_contract"
authors = [""]
compiler_version = "0.1"
type = "contract"

[dependencies]
aztec = { path = "path/to/aztec-packages/yarn-project/noir-libs/noir-aztec" }
```

You are now ready to write your own contracts! 

> You can replace the content of the generated file `example_contract/src/main.nr` with your contract code.

Reviewer: This seems like something that should the starter-kit should handle or aztec cli init or just template on gh.

> You can replace the content of the generated file `example_contract/src/main.nr` with your contract code.

Author: Contract scope replalcement is explained in another link

## Next Steps
You can learn more about writing contracts from the [Contracts section](../contracts/main.md). 

> For now you can use the [PrivateToken contract example here](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/noir-contracts/src/contracts/private_token_contract/src/main.nr)

Reviewer: ```suggestion
For now you can use the [PrivateToken contract example here](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/noir-contracts/src/contracts/private_token_contract/src/main.nr).
```

> After writing the contract, you have to compile it. Details can be found [here](../contracts/compiling.md)

Reviewer: ```suggestion
After writing the contract, you have to compile it. Details can be found [here](../contracts/compiling.md).
```

> After compiling, you can deploy your contract to the Aztec network. Relevant instructions and explainations can be found [here](../contracts/deploying.md)

Reviewer: ```suggestion
After compiling, you can deploy your contract to the Aztec network. Relevant instructions and explainations can be found [here](../contracts/deploying.md).
```
